### PR TITLE
Changes to fulfill CI requirements

### PR DESCRIPTION
**Pull request by llewelld**
This includes various changes to get the code to pass the CI checis:
1. Codeconv version 2.1.12 appears to have been pulled from PyPI for security reasons (see https://github.com/codecov/python-standard/issues/31 ) and so the version has been bumped up to 2.1.13.
2. Skip isort for pre-commit.ci to avoid it conflicting with the pre-commit worker checks.
3. Some additional type checking to get mypi to accept the code.
4. Some code renames, including changing DDPFullyShardedNativeStrategy to FSDPStrategy to account for changes to the Lightning API.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ ci:
   autoupdate_commit_msg: "[pre-commit.ci] pre-commit suggestions"
   autoupdate_schedule: quarterly
   # submodules: true
+  skip: [isort]
```
Comment (Borda) [inline .pre-commit-config.yaml:11]: why is this needed? What is the reason locally it passes but not in CI?

File: app/bench_fsdp_mingpt.py [hunk 2]
```diff
@@ -40,18 +42,28 @@ def create(self) -> Tuple[models.FSDPMinGPT, DataLoader]:
         return model, dataloader
 
     def train(self, model: "LightningModule", dataloader: DataLoader) -> float:
+        assert isinstance(
```
Comment (Borda) [inline app/bench_fsdp_mingpt.py:45]: then rather Exception

File: app/bench_single_mingpt.py [hunk 3]
```diff
@@ -46,17 +48,27 @@ def train(
         model: "LightningModule",
         dataloader: torch.utils.data.DataLoader,
     ) -> Optional[float]:
+        assert isinstance(
```
Comment (Borda) [inline app/bench_single_mingpt.py:51]: maybe wrap it is private function?

**Comment by llewelld** [comment]
Thank you for making changes on this and sorry for my lack of communication; I'm afraid I missed your original comments in my github notifications.

**Review by Borda — APPROVED** [review]

**Pull request merged**